support/macros/tests/tests.rs

```diff
@@ -0,0 +1,7 @@
+use subtensor_macros::freeze_struct;
+
+#[freeze_struct("ecdcaac0f6da589a")]
```

Comment:
How can one calculate this hash in an easy way?

Reply:
people running RA will see it literally in their IDE as well

Reply:
also worth noting it is immune to doc comment changes, they aren't part of the hash code calculation

Reply:
I see, that's cool! but will this work also for generic structs? It seems like it will also break if you rename a generic or field, even if it is the same encoding? 

I think it is certainly a cool idea, and if no similar rust crate exists, we can make it an independent Rust crate, and reference it in our docs.

Reply:
Yeah it could be modified to be immune to name changes. I wasn't sure if that was safe in all cases so I left that out of this one

Reply:
Same goes for generics, whatever you want it to ignore in the hash code calculation can be added to the visitor pattern